src/lwc/geBatchGiftEntryTable/geBatchGiftEntryTable.js

```diff
@@ -150,12 +158,14 @@ export default class GeBatchGiftEntryTable extends LightningElement {
             row[idProperty] === dataRow[idProperty]
         );
 
+        let rows = this.setDatatableRecordsForImperativeCall([dataRow]);
+
         if (existingRowIndex !== -1) {
             this.data.splice(existingRowIndex, 1, dataRow);
             this.data = [...this.data];
         } else {
-            this.data = [dataRow, ...this.data];
-            if (this.hasData == false) {
+            this.data = [...rows, ...this.data];
```

Comment:
If we're going to change this, can we use `Array.concat()` on the two arrays instead of the spread operator? I initially did this with the spread operator but concat is more performant when you don't actually need the spread, which I don't think is needed here.

Reply:
@kenlewis-sfdo if we use 1concat1 on the rows for the table it'll add each new item at the end of the array and not the beginning, which would put the new items at the bottom of the table. Not sure if we want new rows to continue displaying at the top of the table, but I think changing to `concat` would have that side effect of adding at the end.